docs/development/js/js-coding-style.md

```diff
@@ -0,0 +1,66 @@
+# General Javascript Coding Guidelines
+
+## ECMAScript 6
+
+Javascript code must be written in [ES6](http://www.ecma-international.org/ecma-262/6.0/index.html). ES6 runs in all relevant browsers supported by ILIAS. Note: IE11 does not support ES6 and ILIAS does not support IE11 since release 6.0.
+
+## Use Strict
+
+All code files should ensure strict code syntax mode. Some code will automatically run in strict mode, e.g. [ES6 modules or classes](http://www.ecma-international.org/ecma-262/6.0/#sec-strict-mode-code).
+
+All other cases should declare strict mode in the first line of the code file.
+
+```
+'use strict';
+```
+
+## Declare Globals
+
+If global identifiers like `$`or `il` are being used, they should be declared at the top of the file (after `use strict`).
+
+```
+/* global $, il */
+``` 
+
+This will prevent any "... is not defined" errors in the StandardJS syntax checker (see next chapter).
+
+## StandardJS
+
+Javascript code should be written using [StandardJS](https://standardjs.com/rules.html) as coding style.
```

Comment:
Maybe the [Airbnb Coding Standard](https://github.com/airbnb/javascript) could be an alternative worth discussing. According to my frontend colleagues and friends (in other companies), this is the most hyped and (according to them) most comprehensive JavaScript coding style.

Reply:
I'm all for the commitment to a single standard!
Please note that this one e.g. requires to write variables in CamelCase, which is directly oppiste to the code-style for php in ILIAS.
As I said, just a note...